aws-samples/crossplane-aws-blueprints
# add initial tests

### What does this PR do?

Adds initial test implementations for XRDs and GitHub action for PRs. 

This uses kuttl (https://kuttl.dev/). It's mainly used for testing Kubernetes operators and controllers but I think it's a great fit for our repo as well. As you can see, most tests are written in yam files with an option to run bash scripts. 
Currently there is only one test. It will fail if there are any XRD with status "False" for "Established" condition. 

Testing procedure is as follows:
1. Install kind (preinstalled for GitHub actions)
2. Install kuttl. 
3. Run kuttl (`kubectl kuttl test --config tests/kuttl/test-suite.yaml` ). 
4. It creates a kind local cluster. 
5. It will install crossplane using the crossplane helm chart. It will install AWS, Jet AWS, and Kubernetes providers as well. 
6. Install all compositions in this repository. 
7. Run tests. 

In the future, we will need to be able to create actual resources in AWS for better testing ,but I think this is a good start. 

You can see what it looks like when it runs here: https://github.com/nabuskey/crossplane-aws-blueprints/actions/runs/2261439674
